### PR TITLE
Switch to HTTP APIs by default

### PR DESCRIPTION
**Pull request by mnapoli**
The goal is to move towards HTTP APIs instead of REST APIs, at least for the most common scenarios.

This PR updates some examples, and duplicates some other examples. More specifically, it:

- introduces new examples focused on HTTP APIs (they duplicate existing "rest-api" examples, in order to preserve examples about REST APIs)
- update existing examples that were not focused on REST APIs

The goal was to keep examples about REST (not remove them entirely).

Some examples, like "expressjs", "simple HTTP endpoint", etc. were not specifically about REST APIs, so I updated them to use HTTP APIs.

TODO:

- [x] JS
- [x] Typescript
- [x] Python
- [x] Ruby
- [x] Go
- [x] Java
- [x] Rust
- [x] Update the documentation to use HTTP APIs by default: https://github.com/serverless/serverless/pull/9777

### CODE REVIEW

File: aws-node-express-api/README.md [hunk 1]
```diff
@@ -22,21 +22,13 @@ This template configures a single function, `api`, which is responsible for hand
...
-This example is made to work with the Serverless Framework dashboard, which includes advanced features such as CI/CD, monitoring, metrics, etc.
-
-In order to deploy with dashboard, you need to first login with:
-
-```
-serverless login
-```
-
-install dependencies with:
+Install dependencies with:
```
Comment (pgrzesik) [inline aws-node-express-api/README.md:25]: In the section `Anatomy of the template` we should also mention that it now uses `httpApi` and not `http` event type (plus change the reference to proper docs)

File: aws-node-express-dynamodb-api/serverless.yml [hunk 2]
```diff
@@ -48,7 +41,5 @@ resources:
         KeySchema:
           - AttributeName: userId
             KeyType: HASH
-        ProvisionedThroughput:
-          ReadCapacityUnits: 1
-          WriteCapacityUnits: 1
+        BillingMode: PAY_PER_REQUEST
```
Comment (pgrzesik) [inline aws-node-express-dynamodb-api/serverless.yml:44]: good call :+1:

**Review by pgrzesik — COMMENTED** [review]
Looks good in general, good call :+1: I have only one concern, other than that we should be good to go here

File: aws-node-express-api/README.md (hunk 1 above)
Comment (mnapoli) [reply aws-node-express-api/README.md:25]: Oh good call, that's fixed!

File: aws-node-express-dynamodb-api/serverless.yml (hunk 2 above)
Comment (mnapoli) [reply aws-node-express-dynamodb-api/serverless.yml:44]: 👍  some templates had this, some others didn't. I figured we might as well make it consistent over this option, it's best to get started.

**Comment by mnapoli** [comment]
This PR is ready.

After merging this, the next step will be https://github.com/serverless/serverless/pull/9777 (and what's in TODO in that PR). This PR here can be merged and that will unblock the other tasks (@pgrzesik let me know if I'm wrong and we need to merge something simultaneously, but I don't think this is the case, I think we can merge this now and do https://github.com/serverless/serverless/pull/9777 afterwards).

**Pull request merged**